x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/FunctionResolutionStrategy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ql.expression.function;
+
+import org.elasticsearch.xpack.ql.session.Configuration;
+
+
+/**
+ * Strategy indicating the type of resolution to apply for resolving the actual function definition in a pluggable way.
+ */
+public interface FunctionResolutionStrategy {
+
+    /**
+     * Default behavior of standard function calls like {@code ABS(col)}.
+     */
+    FunctionResolutionStrategy DEFAULT = new FunctionResolutionStrategy() {
+    };
+
+    /**
+     * Build the real function from this one and resolution metadata.
+     */
+    default Function buildResolved(UnresolvedFunction uf, Configuration cfg, FunctionDefinition def) {
+        return def.builder().build(uf, false, cfg);
+    }
+
+    /**
+     * The name of the kind of thing being resolved. Used when
```

Comment:
1st sentence might be worth a rephrase. (`What kind of thing is being resolved?`?)